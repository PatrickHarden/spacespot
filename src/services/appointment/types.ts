export interface AppointmentState {
  data?: Array<Appointment>
  error?: string
}

export interface Appointment {
  id: number
  enquireId: number
  description: string
  peer1Id: string
  peer2Id: string
  proposedTime: string
  createdAt: string
  lastUpdatedAt: string
  lastProposedBy: string
  appointmentStatus: string
}

export interface NewAppointment {
  description: string
  proposedTime: string
}
export interface EditAppointment {
  id: string
  description: string
  proposedTime: string
}
export interface AppointmentAction {
  type: string
  payload?: Array<Appointment> | NewAppointment | EditAppointment | string
}
