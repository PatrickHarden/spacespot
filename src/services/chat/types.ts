export interface ChatAction {
  type: string
  payload?: ChatResponse | ChatMessage | string
}
export interface MessageAction {
  type: string
  payload?: string
}
export interface ChatState {
  data?: ChatResponse
  error?: string
}

export interface ChatResponse {
  chatId: number
  enquiryId: number
  peer1Id: string
  peer2Id: string
  messages: Array<ChatMessage>
}

export interface ChatMessage {
  userId: string
  message: string
  sentAt: string
}
