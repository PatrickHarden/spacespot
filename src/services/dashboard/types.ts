import { Space } from 'services/space/types'

export interface DashboardState {
  buildings: Array<Space>
  spaces: { [key: string]: Array<Space> }
  error?: string
  isLoading: boolean
  initOK: boolean
  pendingDelete: Array<string>
}
