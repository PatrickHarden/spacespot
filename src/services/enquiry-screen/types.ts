export interface EnquiryScreenAction {
  type: string
  payload: EnquiryScreenStateData | string
}
export interface EnquiryScreenState {
  data?: EnquiryScreenStateData
  error?: string
}
export interface EnquiryScreenStateData {
  id: string
}
