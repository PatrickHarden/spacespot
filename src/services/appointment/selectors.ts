import { get } from 'lodash'
import moment from 'moment'
import { SpacespotState } from '../global/types'
import { Appointment } from './types'
import { AppointmentProps } from '@cbreenterprise/spacespot-ui'

import userSelectors from '../user/selectors'

const appointment = (state: SpacespotState) => get(state, 'appointment.data')
const appointmentsUser = (state: SpacespotState) => {
  const appointments = appointment(state)
  const userId = userSelectors.accountIdentifier(state)
  return appointments
    ? appointments.map(
        (appoint: Appointment): AppointmentProps => ({
          id: appoint.id.toString(),
          date: moment(appoint.proposedTime).toDate(),
          description: appoint.description,
          appointmentStatus: appoint.appointmentStatus,
          isOwnLast: userId === appoint.lastProposedBy,
        }),
      )
    : []
}
const error = (state: SpacespotState) => get(state, 'appointment.error')

export default {
  appointment,
  appointmentsUser,
  error,
}
