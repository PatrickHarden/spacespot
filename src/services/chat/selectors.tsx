import React from 'react'
import styled from 'styled-components'
import { get } from 'lodash'
import moment from 'moment'
import { ChatMsgProps } from '@cbreenterprise/spacespot-ui'
import { SpacespotState } from '../global/types'
import { ChatMessage } from './types'
import eventSelectors from 'services/event/selectors'
import userSelectors from 'services/user/selectors'
import Avatar from 'components/Avatar/Avatar'

const getChat = (state: SpacespotState) => get(state, 'chat.data')
const getChatId = (state: SpacespotState) => get(getChat(state), 'chatId')
const getEnquiryId = (state: SpacespotState) => get(getChat(state), 'enquiryId')
const getPeer1Id = (state: SpacespotState) => get(getChat(state), 'peer1Id')
const getPeer2Id = (state: SpacespotState) => get(getChat(state), 'peer2Id')
const getMessages = (state: SpacespotState): Array<ChatMessage> =>
  get(getChat(state), 'messages')

const ChatAvatar = styled(Avatar)`
  flex: 0 0 auto;
`

const getAvatar = (id: string, isLandlord: boolean) => {
  return <ChatAvatar userId={id} isLandlord={isLandlord} />
}

const getMessagesUser = (state: SpacespotState): Array<ChatMsgProps> => {
  const accountIdentifier = userSelectors.accountIdentifier(state)
  const isLandlord = userSelectors.isLandlord(state)
  const messages = getMessages(state)
  const enquiryId = getEnquiryId(state)
  const events = eventSelectors.getEvents(enquiryId)(state)

  const eventMsgs = events
    ? events.map(e => {
        const isOwn = e.eventMessage.user === accountIdentifier
        const isLandlordMsg = isLandlord ? isOwn : !isOwn
        return {
          ...e,
          id: e.eventDate.toString(),
          isOwn: isOwn,
          time: moment(e.eventDate).toDate(),
          text: '',
          avatar: () => getAvatar(e.eventMessage.user, isLandlordMsg),
        }
      })
    : []

  const userMsgs = messages
    ? messages
        .map(msg => {
          if (!msg.message || msg.message === '') {
            return undefined
          }
          const isOwn = msg.userId === accountIdentifier
          const isLandlordMsg = isLandlord ? isOwn : !isOwn
          return {
            id: msg.sentAt,
            isOwn: isOwn,
            time: moment(msg.sentAt).toDate(),
            text: msg.message,
            avatar: () => getAvatar(msg.userId, isLandlordMsg),
          }
        })
        .filter(msg => msg !== undefined)
    : []

  const sortedMsgs = eventMsgs
    .concat(userMsgs as Array<any>)
    .sort(function(a, b) {
      return a.time.getTime() - b.time.getTime()
    })
    .map((msg: any) => ({
      ...msg,
      time: moment(msg.time).fromNow(),
    }))
  return sortedMsgs
}

const getError = (state: SpacespotState) => get(state, 'chat.error')

export default {
  getChat,
  getChatId,
  getEnquiryId,
  getPeer1Id,
  getPeer2Id,
  getMessages,
  getMessagesUser,
  getError,
}
