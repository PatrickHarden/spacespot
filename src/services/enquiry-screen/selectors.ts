import { get } from 'lodash'
import { SpacespotState } from '../global/types'

const getId = (state: SpacespotState) => get(state, 'enquiryscreen.data.id')
const getError = (state: SpacespotState): string =>
  get(state, 'enquiryscreen.error')

export default {
  getId,
  getError,
}
