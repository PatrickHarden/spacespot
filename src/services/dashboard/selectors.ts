import { get, chain } from 'lodash'
import { SpacespotState } from 'services/global/types'

import { Space, TranslatePropSpace } from 'services/space/types'
import selectors from 'services/space/selectors'

import { DashboardState } from './types'

const spacesBuildings = (spaces: Array<Space>): Array<TranslatePropSpace> =>
  spaces
    ? spaces
        .map((space: Space) => selectors.translateBuilding(space, true))
        .sort((spaceA, spaceB) => (spaceA.name > spaceB.name ? 1 : -1))
    : []

const selectDashboard = (state: SpacespotState) => get(state, 'dashboard')

const selectDashboardBuildings = (state: SpacespotState) => {
  const spaces = get(selectDashboard(state), 'buildings', [])
  return spacesBuildings(spaces)
}

const selectDashboardSpaceCurried = (state: SpacespotState) => (id: string) =>
  chain(selectDashboard(state))
    .get('spaces')
    .map((building: Array<Space>) =>
      building.filter((space: Space) => get(space, 'Common.PrimaryKey') === id),
    )
    .filter((building: Array<Space>) => building.length > 0)
    .head()
    .head()
    .value()

const selectDashboardBuildingCurried = (state: SpacespotState) => (
  id: string,
) =>
  chain(selectDashboard(state) as DashboardState)
    .get('buildings')
    .filter((build: Space) => get(build, 'Common.PrimaryKey') === id)
    .head()
    .value()

const selectDashboardSpaces = (state: SpacespotState) => {
  const spaces = get(selectDashboard(state), 'spaces')
  if (!spaces) {
    return {}
  }
  return Object.keys(spaces).reduce(
    (
      accumulator: { [key: string]: Array<TranslatePropSpace> },
      id: string,
    ) => ({
      ...accumulator,
      [id]: spacesBuildings(spaces[id] as Array<Space>),
    }),
    {},
  )
}

const isDashboardLoading = (state: SpacespotState): boolean => {
  return get(state, 'dashboard.isLoading', false)
}

const isDashboardInitOK = (state: SpacespotState): boolean => {
  return get(state, 'dashboard.initOK', false)
}

const pendingDelete = (state: SpacespotState): Array<string> => {
  return get(state, 'dashboard.pendingDelete', [])
}

export default {
  selectDashboardBuildings,
  selectDashboardBuildingCurried,
  selectDashboardSpaceCurried,
  selectDashboardSpaces,
  isDashboardLoading,
  isDashboardInitOK,
  pendingDelete,
}
