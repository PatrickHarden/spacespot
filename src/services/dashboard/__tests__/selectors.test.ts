import { SpacespotState } from '../../global/types'
import selectors from '../selectors'
import DashboardBuildingSpace from 'services/space/__mocks__/dashboard-building-space.json'

describe('Dashboard selectors', () => {
  const itemDashboardBuildingSpace: SpacespotState = {
    ...((DashboardBuildingSpace as unknown) as SpacespotState),
  }

  it('selector get spaces data', () => {
    expect(
      selectors.selectDashboardSpaces(itemDashboardBuildingSpace),
    ).toMatchSnapshot()
  })
  it('selector get spaces no data', () => {
    expect(selectors.selectDashboardSpaces({})).toEqual({})
  })
  it('selector get buildings data', () => {
    expect(
      selectors.selectDashboardBuildings(itemDashboardBuildingSpace),
    ).toMatchSnapshot()
  })
  it('selector get buildings data', () => {
    expect(
      selectors.selectDashboardSpaceCurried(itemDashboardBuildingSpace)(
        'SS_SPACE_326',
      ),
    ).toMatchSnapshot()
  })
})
