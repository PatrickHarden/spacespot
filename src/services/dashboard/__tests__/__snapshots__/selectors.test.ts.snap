// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard selectors selector get buildings data 1`] = `
Array [
  Object {
    "address": "Bryggegata, Oslo, Noruega",
    "aspects": Array [
      "hasReception-Manned",
      "hasGym",
    ],
    "availability": undefined,
    "buildingId": "SS_BUILDING_325",
    "city": "Oslo",
    "country": "Norway",
    "description": "building description",
    "district": "",
    "fixedDesks": 0,
    "floor": 0,
    "hotDesks": 0,
    "key": "SS_BUILDING_325",
    "latLng": Object {
      "lat": 59.91041,
      "lng": 0,
    },
    "location": undefined,
    "name": "building headline",
    "photo": "https://uatlistingssearchcbreeuw.blob.core.windows.net/fileassets/SS_BUILDING_325/5cd79f56/11%20Strand%201_Photo_1_small.jpg",
    "postCode": "28004",
    "rent": "",
    "sevicedOffices": 0,
    "size": 0,
    "status": 2,
    "subdistrict": Object {
      "id": 0,
      "name": "",
    },
    "type": "FIXED",
    "uploadedDocs": Array [],
    "uploadedImages": Array [
      Object {
        "Common.ImageCaption": "11 Strand 1_Photo_1_small.jpg",
        "Common.ImageResources": Array [
          Object {
            "Common.Breakpoint": "original",
            "Common.Image.Height": 426,
            "Common.Image.Width": 640,
            "Common.Resource.Uri": "/resources/fileassets/SS_BUILDING_325/5cd79f56/11%20Strand%201_Photo_1_small.jpg",
            "Source.Uri": "https://uatlistingssearchcbreeuw.blob.core.windows.net/fileassets/SS_BUILDING_325/5cd79f56/11%20Strand%201_Photo_1_small.jpg",
          },
          Object {
            "Common.Breakpoint": "small",
            "Common.Image.Height": 426,
            "Common.Image.Width": 640,
            "Common.Resource.Uri": "/resources/fileassets/SS_BUILDING_325/5cd79f56/11%20Strand%201_Photo_1_small_Photo_1_small.jpg",
            "Source.Uri": "https://uatlistingssearchcbreeuw.blob.core.windows.net/fileassets/SS_BUILDING_325/5cd79f56/11%20Strand%201_Photo_1_small.jpg",
          },
          Object {
            "Common.Breakpoint": "medium",
            "Common.Image.Height": 852,
            "Common.Image.Width": 1280,
            "Common.Resource.Uri": "/resources/fileassets/SS_BUILDING_325/5cd79f56/11%20Strand%201_Photo_1_small_Photo_1_medium.jpg",
            "Source.Uri": "https://uatlistingssearchcbreeuw.blob.core.windows.net/fileassets/SS_BUILDING_325/5cd79f56/11%20Strand%201_Photo_1_small.jpg",
          },
          Object {
            "Common.Breakpoint": "large",
            "Common.Image.Height": 1065,
            "Common.Image.Width": 1600,
            "Common.Resource.Uri": "/resources/fileassets/SS_BUILDING_325/5cd79f56/11%20Strand%201_Photo_1_small_Photo_1_large.jpg",
            "Source.Uri": "https://uatlistingssearchcbreeuw.blob.core.windows.net/fileassets/SS_BUILDING_325/5cd79f56/11%20Strand%201_Photo_1_small.jpg",
          },
        ],
      },
    ],
  },
]
`;

exports[`Dashboard selectors selector get buildings data 2`] = `
Object {
  "Common.ActualAddress": Object {
    "Common.Country": "Norway",
    "Common.PostCode": "28004",
    "Common.PostalAddresses": Array [
      Object {
        "Common.Language": "en-GB",
        "Common.Line1": "Space headline",
        "Common.Line2": "Bryggegata, Oslo, Noruega",
        "Common.Locallity": "Oslo",
      },
    ],
  },
  "Common.Availability": Object {
    "Common.AvailabilityDate": "2019-12-18",
    "Common.AvailabilityKind": "AvailableFromKnownDate",
  },
  "Common.AvailableFrom": "2019-12-18",
  "Common.Brochures": Array [],
  "Common.BuildingOperatorID": "920d0a43-7322-43b1-a22c-01437939c880",
  "Common.Charges": Array [
    Object {
      "Common.Amount": 21,
      "Common.AmountKind": "Value",
      "Common.ChargeKind": "Rent",
      "Common.ChargeModifer": "Fixed",
      "Common.CurrencyCode": "NOK",
      "Common.Interval": "Monthly",
      "Common.PerUnit": "sqm",
      "Common.TaxModifer": "PlusVAT",
    },
    Object {
      "Common.Amount": 1200,
      "Common.AmountKind": "Value",
      "Common.ChargeKind": "ServiceCharge",
      "Common.ChargeModifer": "Fixed",
      "Common.CurrencyCode": "NOK",
      "Common.Interval": "Monthly",
      "Common.PerUnit": "sqm",
      "Common.TaxModifer": "PlusVAT",
    },
  ],
  "Common.ContactGroup": Object {
    "Common.GroupType": "AreaOffice",
  },
  "Common.Coordinate": Object {
    "lat": 59.91041,
    "lon": 0,
  },
  "Common.FitOutOptions": Array [
    Object {
      "Common.Charges": Array [
        Object {
          "Common.Amount": 12,
          "Common.AmountKind": "Value",
          "Common.ChargeKind": "PropertyAccommodation",
          "Common.ChargeModifer": "Fixed",
          "Common.CurrencyCode": "NOK",
          "Common.Interval": "Monthly",
          "Common.OnApplication": true,
          "Common.TaxModifer": "PlusVAT",
        },
      ],
      "Common.Description": Array [
        Object {
          "Common.CultureCode": "es-ES",
          "Common.Text": "head desc fit1",
        },
      ],
      "Common.Identifier": "fitout",
      "Common.Name": Array [
        Object {
          "Common.CultureCode": "es-ES",
          "Common.Text": "head fitout 1",
        },
      ],
      "Common.SpaceID": "SS_SPACE_438",
    },
  ],
  "Common.FloorPlans": Array [],
  "Common.Highlights": Array [
    Object {
      "Common.Highlight": Array [
        Object {
          "Common.CultureCode": "en-GB",
          "Common.Text": "Space headline",
        },
      ],
    },
  ],
  "Common.HomeSite": "no-comm",
  "Common.IsParent": false,
  "Common.LastProcessed": "2019-12-17T11:41:18.1349463Z",
  "Common.LastUpdated": "2019-12-17T11:39:19.8056303Z",
  "Common.MinimumSize": Array [
    Object {
      "Common.Size": 500,
      "Common.Units": "sqm",
    },
  ],
  "Common.ParentProperty": "SS_BUILDING_325",
  "Common.Photos": Array [
    Object {
      "Common.ImageCaption": "40 Villiers Street.External_Photo_1_small.jpg",
      "Common.ImageResources": Array [
        Object {
          "Common.Breakpoint": "original",
          "Common.Image.Height": 492,
          "Common.Image.Width": 640,
          "Common.Resource.Uri": "/resources/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small.jpg",
          "Source.Uri": "https://uatlistingssearchcbreeuw.blob.core.windows.net/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small.jpg",
        },
        Object {
          "Common.Breakpoint": "small",
          "Common.Image.Height": 492,
          "Common.Image.Width": 640,
          "Common.Resource.Uri": "/resources/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small_Photo_1_small.jpg",
          "Source.Uri": "https://uatlistingssearchcbreeuw.blob.core.windows.net/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small.jpg",
        },
        Object {
          "Common.Breakpoint": "medium",
          "Common.Image.Height": 984,
          "Common.Image.Width": 1280,
          "Common.Resource.Uri": "/resources/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small_Photo_1_medium.jpg",
          "Source.Uri": "https://uatlistingssearchcbreeuw.blob.core.windows.net/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small.jpg",
        },
        Object {
          "Common.Breakpoint": "large",
          "Common.Image.Height": 1230,
          "Common.Image.Width": 1600,
          "Common.Resource.Uri": "/resources/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small_Photo_1_large.jpg",
          "Source.Uri": "https://uatlistingssearchcbreeuw.blob.core.windows.net/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small.jpg",
        },
      ],
    },
  ],
  "Common.PrimaryKey": "SS_SPACE_326",
  "Common.Sizes": Array [
    Object {
      "Common.Dimensions": Array [
        Object {
          "Common.Amount": 500,
          "Common.DimensionsUnits": "sqm",
        },
      ],
      "Common.SizeKind": "MinimumSize",
    },
    Object {
      "Common.Dimensions": Array [
        Object {
          "Common.Amount": 1000,
          "Common.DimensionsUnits": "sqm",
        },
      ],
      "Common.SizeKind": "TotalSize",
    },
  ],
  "Common.SourceLastupdated": "2019-12-17T11:39:19.656606Z",
  "Common.TotalSize": Array [
    Object {
      "Common.Size": 1000,
      "Common.Units": "sqm",
    },
  ],
  "Common.UsageType": "Office",
  "Common.Walkthrough": "http://www.matterports.es",
  "Common.Website": "https://www.spacespot.com",
  "Dynamic.DevelopmentOpportunity": false,
  "Dynamic.FlexCharge": null,
  "Dynamic.Furnished": "unfurnished",
  "Dynamic.Has24HourAccess": false,
  "Dynamic.HasAccessControl": false,
  "Dynamic.HasBalcony": false,
  "Dynamic.HasBarCafeteria": false,
  "Dynamic.HasBicycleRack": false,
  "Dynamic.HasBreakoutSpace": false,
  "Dynamic.HasBusinessLounge": false,
  "Dynamic.HasCleaning": false,
  "Dynamic.HasCoworkingArea": false,
  "Dynamic.HasDoorkeeper": false,
  "Dynamic.HasFixedDesks": false,
  "Dynamic.HasFloorsRaised": false,
  "Dynamic.HasFreeMeetingRooms": false,
  "Dynamic.HasGarden": false,
  "Dynamic.HasGym": false,
  "Dynamic.HasHotDesks": false,
  "Dynamic.HasLift": false,
  "Dynamic.HasParking": false,
  "Dynamic.HasReception": false,
  "Dynamic.HasRoofTerrace": false,
  "Dynamic.HasServicedOffices": false,
  "Dynamic.HasShowers": false,
  "Dynamic.HasStorageArchives": false,
  "Dynamic.HasSwimmingPool": false,
  "Dynamic.HasTeaCoffee": false,
  "Dynamic.HasTerrace": false,
  "Dynamic.HasWheelchairAccess": false,
  "Dynamic.HasWheelchairAccessToilets": false,
  "Dynamic.IsPenthouse": false,
  "Dynamic.LetUnderOffer": false,
  "Dynamic.LettingCharge": 0.19,
  "Dynamic.MaxArea": null,
  "Dynamic.MinArea": 5381.957525591208,
  "Dynamic.MinDesks": 5381.957525591208,
  "Dynamic.MinimumSize": Array [
    Object {
      "Common.Size": 500,
      "Common.Units": "sqm",
    },
  ],
  "Dynamic.NetCharge": null,
  "Dynamic.NewHome": false,
  "Dynamic.PrimaryImage": Object {
    "Common.ImageCaption": "40 Villiers Street.External_Photo_1_small.jpg",
    "Common.ImageResources": Array [
      Object {
        "Common.Breakpoint": "original",
        "Common.Image.Height": 492,
        "Common.Image.Width": 640,
        "Common.Resource.Uri": "/resources/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small.jpg",
        "Source.Uri": "https://uatlistingssearchcbreeuw.blob.core.windows.net/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small.jpg",
      },
      Object {
        "Common.Breakpoint": "small",
        "Common.Image.Height": 492,
        "Common.Image.Width": 640,
        "Common.Resource.Uri": "/resources/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small_Photo_1_small.jpg",
        "Source.Uri": "https://uatlistingssearchcbreeuw.blob.core.windows.net/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small.jpg",
      },
      Object {
        "Common.Breakpoint": "medium",
        "Common.Image.Height": 984,
        "Common.Image.Width": 1280,
        "Common.Resource.Uri": "/resources/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small_Photo_1_medium.jpg",
        "Source.Uri": "https://uatlistingssearchcbreeuw.blob.core.windows.net/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small.jpg",
      },
      Object {
        "Common.Breakpoint": "large",
        "Common.Image.Height": 1230,
        "Common.Image.Width": 1600,
        "Common.Resource.Uri": "/resources/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small_Photo_1_large.jpg",
        "Source.Uri": "https://uatlistingssearchcbreeuw.blob.core.windows.net/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small.jpg",
      },
    ],
  },
  "Dynamic.SalePrice": null,
  "Dynamic.TotalArea": 10763.915051182415,
  "Dynamic.TotalDesks": 10763.915051182415,
  "Dynamic.TotalSize": Array [
    Object {
      "Common.Size": 1000,
      "Common.Units": "sqm",
    },
  ],
  "Dynamic.UnderOffer": false,
}
`;

exports[`Dashboard selectors selector get spaces data 1`] = `
Object {
  "SS_BUILDING_325": Array [
    Object {
      "address": "Bryggegata, Oslo, Noruega",
      "aspects": undefined,
      "availability": 2019-12-18T00:00:00.000Z,
      "buildingId": "SS_SPACE_326",
      "city": "Oslo",
      "country": "Norway",
      "description": "",
      "district": "",
      "fixedDesks": 0,
      "floor": 0,
      "hotDesks": 0,
      "key": "SS_SPACE_326",
      "latLng": Object {
        "lat": 59.91041,
        "lng": 0,
      },
      "location": undefined,
      "name": "Space headline",
      "photo": "https://uatlistingssearchcbreeuw.blob.core.windows.net/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small.jpg",
      "postCode": "28004",
      "rent": "21kr",
      "sevicedOffices": 0,
      "size": 500,
      "status": 2,
      "subdistrict": Object {
        "id": 0,
        "name": "",
      },
      "type": "FIXED",
      "uploadedDocs": Array [],
      "uploadedImages": Array [
        Object {
          "Common.ImageCaption": "40 Villiers Street.External_Photo_1_small.jpg",
          "Common.ImageResources": Array [
            Object {
              "Common.Breakpoint": "original",
              "Common.Image.Height": 492,
              "Common.Image.Width": 640,
              "Common.Resource.Uri": "/resources/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small.jpg",
              "Source.Uri": "https://uatlistingssearchcbreeuw.blob.core.windows.net/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small.jpg",
            },
            Object {
              "Common.Breakpoint": "small",
              "Common.Image.Height": 492,
              "Common.Image.Width": 640,
              "Common.Resource.Uri": "/resources/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small_Photo_1_small.jpg",
              "Source.Uri": "https://uatlistingssearchcbreeuw.blob.core.windows.net/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small.jpg",
            },
            Object {
              "Common.Breakpoint": "medium",
              "Common.Image.Height": 984,
              "Common.Image.Width": 1280,
              "Common.Resource.Uri": "/resources/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small_Photo_1_medium.jpg",
              "Source.Uri": "https://uatlistingssearchcbreeuw.blob.core.windows.net/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small.jpg",
            },
            Object {
              "Common.Breakpoint": "large",
              "Common.Image.Height": 1230,
              "Common.Image.Width": 1600,
              "Common.Resource.Uri": "/resources/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small_Photo_1_large.jpg",
              "Source.Uri": "https://uatlistingssearchcbreeuw.blob.core.windows.net/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small.jpg",
            },
          ],
        },
      ],
    },
  ],
}
`;
